### ISSUE TITLE
S3240 false positive on nested (potential) condtional operators

### ISSUE THREAD

**Issue by Corniel** [SonarSource/sonar-dotnet#3468]
### Description
Rule S3240 suggest when a conditional operator can be used, also when one of the if-branches already contains a conditional operator. When applied, you get a nested conditional operator, which is not okay according to S3358. 

### Repro steps
``` C#
public static class S3240
{
    public static int NoConditionalOperator(int value)
    {
        if (value % 2 == 0) // none-compliant. Will lead to nested conditionals.
        {
            return value == 2 ? 1 : 0;
        }
        else
        {
            return 17;
        }
    }
}
```

### Related information
* v8.10

**Comment by pavel-mikula-sonarsource** [comment]
Hi @Corniel ,

Thank you for reporting this. I'm confirming this as FP. It's indeed not desirable to have nested ternary operators or ternary operator with some really complex return values.

**Comment by haraman21** [comment]
While if-else block already contains an interpolated string with conditional operator, there is an invalid suggestion S3240 to use '?:' and if suggestion is applied it reports S3358 violation for 'Ternary operators should not be nested'
```
string s1 = string.Empty;
if (condition1) // S3240: Use the '?:' operator here
{
    s1 = "some value";
}
else
{
    s1 = $"other value {(condition2 ? "suffix1" : "suffix2")}";
}
```

**Comment by pavel-mikula-sonarsource** [comment]
Hi @haraman21, 

please create new issue here https://community.sonarsource.com/c/rules-and-languages/fp/7 as this one is already closed.

Thank you

**Comment by haraman21** [comment]
> Hi [@haraman21](https://github.com/haraman21),
> 
> please create new issue here https://community.sonarsource.com/c/rules-and-languages/fp/7 as this one is already closed.
> 
> Thank you

Created: https://community.sonarsource.com/t/s3240-false-positive-for-interpolated-string-with-condtional-operator/134362
